config/v1/types_authentication.go

```diff
@@ -262,6 +262,31 @@ type TokenClaimMappings struct {
 	// groups for the cluster identity.
 	// The referenced claim must use array of strings values.
 	Groups PrefixedClaimMapping `json:"groups,omitempty"`
+
+	// uid is an optional field for configuring the claim mapping
+	// used to construct the uid for the cluster identity.
+	//
+	// When using uid.claim to specify the claim it must be a single string value.
+	// When using uid.expression the expression must result in a single string value.
+	//
+	// When omitted, this means the user has no opinion and the platform
+	// is left to choose a default, which is subject to change over time.
+	// The current default is to use the 'sub' claim.
+	//
+	// +optional
+	UID TokenClaimOrExpressionMapping `json:"uid,omitempty"`
```

Comment:
This omitempty won't work, are you relying on it? I believe you are, and therefore need a pointer here

Reply:
I was following the existing fields. Happy to make this a pointer. Not actually relying one this being a pointer though. 

If this field is not provided by user, the system will check for the empty values of the type fields and default accordingly.

This is also a config API so generally we shouldn't be using `omitempty` for discoverability but I felt that consistency with the rest of the API may be best here.

Reply:
> Not actually relying one this being a pointer though.

If the user doesn't specify anything, this will be marshalled with an empty `claim` right? Assuming claim stays as allowing an empty value, then that's fine, but if we change that, we might need to change this too

I would drop the `omitempty` personally here, it doesn't do anything and it feels more wrong to have it

Reply:
I'm going to change this to a pointer. Based on the discussion in https://github.com/openshift/api/pull/2234#discussion_r2007541493 I'm going to make the new struct a fully validated version that we converge all the fields to in the future when all supported management cluster versions support ratcheting validations. This means we will not allow the empty `claim` value.